packages/wells/src/client/api/wellsApi.ts

```diff
@@ -131,12 +142,33 @@ export class WellsAPI extends ConfigureAPI {
       userFilter.polygon = undefined;
       filter = this.convertToApiWellFilter(userFilter);
     }
-    const path = this.getPath('/wells/list', cursor)
-    return await this.client?.asyncPost<WellItems>(path, {'data': filter})
-      .then(response => this.addLazyToAllWellItems(response.data))
-      .catch(err => {
-        throw new HttpError(err.status, err.errorMessage, {})
-    });
+    return filter
+  }
+
+  public filterSlow = async (userFilter: WellFilter): Promise<Well[] | undefined> => {
+    let wells: Well[] = []
+    let cursor = undefined
+    const filter: WellFilterAPI = this.wrapFilter(userFilter);
+    do {
+        const wellsChunk: WellItems | undefined = await this.sendFilterRequest(filter, cursor);
```

Comment:
Can sendFilterRequest return undefined? :open_mouth:

Reply:
yes, since the `Promise` can throw an error, it can return an `undefined`. Typescript complains if it is not added.